packages/reporters/dev-server/src/HMRServer.js

```diff
@@ -46,32 +47,24 @@ export default class HMRServer {
   }
 
   start(): any {
-    let websocketOptions = {
-      /*verifyClient: info => {
-          if (!this.options.host) return true;
-
-          let originator = new URL(info.origin);
-          return this.options.host === originator.hostname;
-        }*/
-    };
-    if (this.options.devServer) {
-      websocketOptions.server = this.options.devServer;
-    } else if (this.options.port) {
-      websocketOptions.port = this.options.port;
-    }
-    this.wss = new WebSocket.Server(websocketOptions);
+    this.wss = new WebSocket.Server(
+      this.options.devServer
+        ? {server: this.options.devServer}
+        : {port: this.options.port},
+    );
 
     this.wss.on('connection', ws => {
-      ws.onerror = this.handleSocketError;
-
       if (this.unresolvedError) {
         ws.send(JSON.stringify(this.unresolvedError));
       }
     });
 
+    // $FlowFixMe[incompatible-call]
     this.wss.on('error', this.handleSocketError);
 
-    return this.wss._server.address().port;
+    let address = this.wss.address();
+    invariant(typeof address === 'object' && address != null);
```

Comment:
Is this check related to updating the `ws` package or just something that was overlooked before ?

Reply:
I moved this to use the public api, which seems to return non-objects for things like unix sockets or if it's not managing a server, I believe. But our use should always have it managing a tcp server.